modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/portlet/action/ImportSystemDataSetMVCResourceCommand.java

```diff
@@ -87,10 +103,205 @@ protected void doTransactionalCommand(
 			).build(),
 			new ServiceContext());
 
+		ObjectDefinition dataSetActionObjectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinition(
+				companyId, "DataSetAction");
+
+		FDSCreationMenu fdsCreationMenu =
+			_fdsCreationMenuRegistry.getFDSCreationMenu(fdsName);
+
+		if (fdsCreationMenu != null) {
+			_addCreationDataSetActionObjectEntries(
+				objectEntry.getObjectEntryId(),
+				dataSetActionObjectDefinition.getDefaultLanguageId(),
+				fdsCreationMenu, _portal.getHttpServletRequest(resourceRequest),
+				_portal.getHttpServletResponse(resourceResponse),
+				dataSetActionObjectDefinition.getObjectDefinitionId());
+		}
+
+		FDSItemActionList fdsItemActionList =
+			_fdsItemActionListRegistry.getFDSItemActionList(fdsName);
+
+		if (fdsItemActionList != null) {
+			_addItemDataSetActionObjectEntries(
+				objectEntry.getObjectEntryId(),
+				dataSetActionObjectDefinition.getDefaultLanguageId(),
+				fdsItemActionList,
+				_portal.getHttpServletRequest(resourceRequest),
+				_portal.getHttpServletResponse(resourceResponse),
+				dataSetActionObjectDefinition.getObjectDefinitionId());
+		}
+
 		JSONPortletResponseUtil.writeJSON(
 			resourceRequest, resourceResponse, objectEntry);
 	}
 
+	private void _addCreationDataSetActionObjectEntries(
+			long dataSetId, String defaultLanguageId,
+			FDSCreationMenu fdsCreationMenu,
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse, long objectDefinitionId)
+		throws Exception {
+
+		CreationMenu creationMenu = fdsCreationMenu.getCreationMenu(
+			httpServletRequest, httpServletResponse);
+
+		List<DropdownItem> primaryDropdownItems =
+			(List<DropdownItem>)creationMenu.get("primaryItems");
+
+		for (DropdownItem dropdownItem : primaryDropdownItems) {
+			Map<String, Serializable> objectEntryValues =
+				HashMapBuilder.<String, Serializable>put(
+					"icon", () -> _getOptionalValue(dropdownItem.get("icon"))
+				).put(
+					"label_i18n",
+					() -> _getLocalizeableValue(
+						defaultLanguageId,
+						_getOptionalValue(dropdownItem.get("label")))
+				).put(
+					"r_dataSetToDataSetActions_l_dataSetId", dataSetId
+				).put(
+					"target", String.valueOf(dropdownItem.get("target"))
+				).put(
+					"type", "creation"
+				).put(
+					"url", () -> _getOptionalValue(dropdownItem.get("href"))
+				).build();
```

Comment:
@dsanz I was wrong in saying PR is somewhat duplicating solution from your https://github.com/liferay-frontend/liferay-portal/pull/4669. We are doing opposite here, `CreationMenu` to `ObjectEntry`, while the other PR does `ObjectEntry` to `CreationMenu`.  "deserialization", if you wish.